File: 002-mockowanie/PHP/tests/Util/EmailTest.php
<?php
declare(strict_types=1);

namespace App\Test\Util;

use App\Util\Email;
use PHPUnit\Framework\TestCase;
use \Exception;

/**
 * Class EmailTest
 * @package App\Test\Util
 */
class EmailTest extends TestCase
{

    /**
     * @var
     */
    private $email;

    /**
     *
     */
    public function setUp(): void
    {
        $this->email = new Email();
    }

    /**
     * @test
     */
    public function send(): void
    {
        $this->assertTrue($this->email->send('kontakt@lukaszstaniszewski.pl'));
    }

    /**
     * @test
     */
    public function emailInvalid(): void
    {
        $this->expectException(Exception::class);
        $this->email->send('kontakt*lukaszstaniszewski.pl');
    }
}

File: 002-mockowanie/Java/src/test/java/lukaszstaniszewski/util/EmailTest.java
package lukaszstaniszewski.util;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertTrue;

/**
 * The type Email test.
 */
public class EmailTest {

    private Email email;

    /**
     * Sets up.
     *
     * @throws Exception the exception
     */
    @Before
    public void setUp() throws Exception {
        this.email = new Email();
    }

    /**
     * Send true.
     *
     * @throws Exception the exception
     */
    @Test
    public void sendTrue() throws Exception {
        assertTrue(this.email.send("kontakt@lukaszstaniszewski.pl"));
    }

    /**
     * Send false.
     *
     * @throws Exception the exception
     */
    @Test(expected = Exception.class)
    public void sendFalse() throws Exception {
        this.email.send("kontaktlukaszstaniszewski.pl");
    }
}
